degree-importer/degree-importer.php

```diff
@@ -0,0 +1,595 @@
+            $post_data  = $post['post_data'];
+            $post_meta  = $post['post_meta'];
+            $post_terms = $post['post_terms'];
+
+            $degree_id = isset( $post_meta['degree_id'] ) ? $post_meta['degree_id'] : null;
+            $degree_type_id = isset( $post_meta['degree_type_id'] ) ? $post_meta['degree_type_id'] : null;
+
+            $post_id = $this->process_post( $post_data, $degree_id, $degree_type_id );
+            $this->process_post_meta( $post_id, $post_meta );
+            $this->process_post_terms( $post_id, $post_terms );
+
+            $import_progress->tick();
+        }
+
+        $import_progress->finish();
+    }
+
+    /**
+     * Cleans a name to make it easier to string compare
+     * @author Jim Barnes
+     * @since 1.0.0
+     * @return string | The cleaned string
+     **/ 
+    private function clean_name( $name ) {
+        $blacklist = array( 'degree', 'program' );
+        $name = strtolower( html_entity_decode( $name, ENT_NOQUOTES, 'UTF-8' ) );
+        $name = str_replace( $blacklist, '', $name );
+        $name = preg_replace( '/[^a-z0-9]/', '', $name );
+
+        return $name;
+    }
+
+    /**
+     * Returns the program type
+     * @author Jim Barnes
+     * @since 1.0.0
+     * @param $type string | The type from the search service
+     * @param $graduate int | The graduate value from the search service
+     * @return string | The newly formatted type
+     **/
+    private function get_program_type( $type, $graduate ) {
+        switch( $type ) {
+            case 'major':
+                if ( $graduate === 0 ) {
+                    $type = 'Undergraduate Degree';
+                } else {
+                    $type = 'Graduate Degree';
+                }
+                break;
+            case 'articulated':
+            case 'accelerated':
+                $type = ucwords( $type ) . ' Program';
+                break;
+            default:
+                $type = ucwords( $type );
+                break;
+        }
+
+        return $type;
+    }
+
+    /**
+     * Returns the undergraduate catalog type
+     * @author Jim Barnes
+     * @since 1.0.0
+     * @param $type string | The search service type
+     * @return string | The undergraduate catalog type.
+     **/
+    private function get_uc_program_type( $type ) {
+        if ( $type === 'major' ) {
+            return 'Degree Program';
+        } else {
+            return $type;
+        }
+    }
+
+    /**
+     * Returns the suffix for the post_name
+     * @author Jim Barnes
+     * @since 1.0.0
+     * @param $name string | The program name
+     * @param $type string | The program type
+     * @param $graduate int | If the program is a graduate program
+     * @return string | The program suffix
+     **/
+    private function get_program_suffix( $name, $type, $graduate ) {
+        $lower_name = strtolower( $name );
+
+        switch( $type ) {
+            case 'minor':
+                return '-minor';
```

Comment:
Do we have a plan for redirecting existing minors and certificate urls without the post_name suffixes after a new import with this logic is done?